Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -498,10 +541,230 @@ private async Task<IEnumerable<MatchViewModel>> ScheduleLadderAsync(int seriesID
+                        day = day.AddMinutes(schedule.EstMatchTime + 15);
+                    }
+                    while (!schedule.AvailableDays[dayOfWeek] || day.AddMinutes(schedule.EstMatchTime + 15) > scheduleEndTime)
+                    {
+                        day = day.AddDays(1);
+                        day = new DateTime(day.Year, day.Month, day.Day).Add(start);
+                        scheduleEndTime = scheduleEndTime.AddDays(1);
+
+                        dayOfWeek = day.ConvertFromUtc(Time_Zones.EST).DayOfWeek.ToString();
+                        surfaceIndex = 0;
+                    }
+
+                    var createdMatch = await _matchService.PostMatchAsync(new MatchUploadViewModel
+                    {
+                        StartTime = day,
+                        SeriesID = series.ID,
+                        SurfaceID = availableSurfaces[surfaceIndex].SurfaceID, 
+                        TeamIDs = new List<int>().AsEnumerable() //no teams
+                    });
+                    createdMatches.Add(createdMatch);
+
+                    var matchBracketPlacement = new MatchBracket()
+                    {
+                        MatchID = createdMatch.ID,
+                        RoundNumber = roundNumber,
+                        RoundOf = numOfBracketParticipatingTeams,
+                        SeedIndex = i,
+                        IsLosers = true
+                    };
+                    await _context.MatchBracket.AddAsync(matchBracketPlacement);
+                    await _context.SaveChangesAsync();
+                    surfaceIndex++;
+                }
+                numOfBracketParticipatingTeams /= (j % 2 == 0 ? 2 : 1);
+                j++;
+                roundNumber++;
+            }
+
+            // GRAND FINALS (played on the same day [if possible] as the losers bracket)
+            for (int i = 1; i < 3; i++)
+            {
+                if (surfaceIndex == availableSurfaces.Length)
+                {
+                    surfaceIndex = 0;
+                    day = day.AddMinutes(schedule.EstMatchTime + 15);
+                }
+                while (!schedule.AvailableDays[dayOfWeek] || day.AddMinutes(schedule.EstMatchTime + 15) > scheduleEndTime)
+                {
+                    day = day.AddDays(1);
+                    day = new DateTime(day.Year, day.Month, day.Day).Add(start);
+                    scheduleEndTime = scheduleEndTime.AddDays(1);
+
+                    dayOfWeek = day.ConvertFromUtc(Time_Zones.EST).DayOfWeek.ToString();
+                    surfaceIndex = 0;
+                }
+                var createdMatch = await _matchService.PostMatchAsync(new MatchUploadViewModel
+                {
+                    StartTime = day,
+                    SeriesID = series.ID,
+                    SurfaceID = availableSurfaces[surfaceIndex].SurfaceID, 
+                    TeamIDs = new List<int>().AsEnumerable() //no teams
```

Comment:
```suggestion
                    TeamIDs = Enumerable.Empty<int>() //no teams
```

Reply:
Interesting methodology. I could implement this, but this would also mean I have to change all existing `list.add()` methods to `____.append()`. Is there any main benefit of using `Enumerable.Empty`?